pkg/tsdb/cloudmonitoring/resource_handler.go

```diff
@@ -27,21 +26,29 @@ type processResponse func(body []byte) ([]json.RawMessage, string, error)
 
 func (s *Service) newResourceMux() *http.ServeMux {
 	mux := http.NewServeMux()
-	mux.HandleFunc("/gceDefaultProject", getGCEDefaultProject)
+	mux.HandleFunc("/gceDefaultProject", s.getGCEDefaultProject())
 	mux.HandleFunc("/metricDescriptors/", s.handleResourceReq(cloudMonitor, processMetricDescriptors))
 	mux.HandleFunc("/services/", s.handleResourceReq(cloudMonitor, processServices))
 	mux.HandleFunc("/slo-services/", s.handleResourceReq(cloudMonitor, processSLOs))
 	mux.HandleFunc("/projects", s.handleResourceReq(resourceManager, processProjects))
 	return mux
 }
 
-func getGCEDefaultProject(rw http.ResponseWriter, req *http.Request) {
-	project, err := utils.GCEDefaultProject(req.Context())
-	if err != nil {
-		writeResponse(rw, http.StatusBadRequest, fmt.Sprintf("unexpected error %v", err))
-		return
+func (s *Service) getGCEDefaultProject() func(rw http.ResponseWriter, req *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
```

Comment:
you don't have any parameters so you don't need the extra wrapper. You can just define:

```
func (s *Service) getGCEDefaultProject(rw http.ResponseWriter, req *http.Request) {
```